### PR TITLE
[CLOUDTRUST-2654] Fix Sonar code smells

### PR DESCRIPTION
**Pull request by fperot74**

### CODE REVIEW

File: keycloak-rest-api-extensions/src/main/java/io/cloudtrust/keycloak/services/resource/api/admin/UserResource.java [hunk 1]
```diff
@@ -1,23 +1,12 @@
...
 
 public class UserResource extends org.keycloak.services.resources.admin.UserResource {
-
-    private AdminPermissionEvaluator auth;
-
-    private AdminEventBuilder adminEvent;
-    private UserModel user;
-
     public UserResource(KeycloakSession session, UserModel user, AdminPermissionEvaluator auth, AdminEventBuilder adminEvent) {
         super(session.getContext().getRealm(), user, auth, adminEvent);
```
Comment (fperot74) [inline keycloak-rest-api-extensions/src/main/java/io/cloudtrust/keycloak/services/resource/api/admin/UserResource.java:10]: Why do we have this class ? It just overrides UserResource with bringing any new feature... May be we should remove it?

File: keycloak-rest-api-extensions/src/main/java/io/cloudtrust/keycloak/services/resource/api/admin/AdminRoot.java [hunk 2]
```diff
@@ -29,15 +31,16 @@ public AdminRoot(KeycloakSession session) {
...
-        if (request.getHttpMethod().equals(HttpMethod.OPTIONS)) {
+        if (HttpMethod.OPTIONS.equals(request.getHttpMethod())) {
             return new AdminCorsPreflightService(request);
         }
 
         AdminAuth auth = authenticateRealmAdminRequest(request.getHttpHeaders());
-        if (auth != null) {
-            logger.debug("authenticated admin access for: " + auth.getUser().getUsername());
+        if (auth == null) {
+            throw new CloudtrustRuntimeException("Can't get AdminAuth");
```
Comment (lagess) [inline keycloak-rest-api-extensions/src/main/java/io/cloudtrust/keycloak/services/resource/api/admin/AdminRoot.java:40]: Shouln't we throw some kind of "UnauthorizedException" (to have a 401, and not a 500)?

File: keycloak-rest-api-extensions/src/main/java/io/cloudtrust/keycloak/services/resource/api/admin/UserResource.java (hunk 1 above)
Comment (lagess) [reply keycloak-rest-api-extensions/src/main/java/io/cloudtrust/keycloak/services/resource/api/admin/UserResource.java:10]: You meant "without bringing", right? Indeed, we should remove it.

File: keycloak-rest-api-extensions/src/main/java/io/cloudtrust/keycloak/services/resource/api/admin/AdminRoot.java (hunk 2 above)
Comment (fperot74) [reply keycloak-rest-api-extensions/src/main/java/io/cloudtrust/keycloak/services/resource/api/admin/AdminRoot.java:40]: I did not want to make functional changes: in previous version, a NullPointerException would have been thrown further (when calling auth.getToken())
But you're right, we can fix this with an UnauthorizedException which would make sense

File: keycloak-rest-api-extensions/src/main/java/io/cloudtrust/keycloak/services/resource/api/admin/UserResource.java (hunk 1 above)
Comment (fperot74) [reply keycloak-rest-api-extensions/src/main/java/io/cloudtrust/keycloak/services/resource/api/admin/UserResource.java]: Removed

File: keycloak-rest-api-extensions/src/main/java/io/cloudtrust/keycloak/services/resource/api/admin/AdminRoot.java (hunk 2 above)
Comment (fperot74) [reply keycloak-rest-api-extensions/src/main/java/io/cloudtrust/keycloak/services/resource/api/admin/AdminRoot.java]: Throwing NotAuthorizedException (UnauthorizedException is deprecated)

**Pull request merged**